config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +113,199 @@ const (
 	// UserDefinedNamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
 	UserDefinedNamespaceIsolated UserDefinedMode = "NamespaceIsolated"
 )
+
+// alertmanagerConfig provides configuration options for the default Alertmanager instance
+// +kubebuilder:validation:XValidation:rule="self.deploymentMode == 'CustomConfig' ? has(self.customConfig) && !has(self.disabled) : true",message="when deploymentMode is CustomConfig, customConfig must be set and disabled must be unset"
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+type AlertmanagerConfig struct {
+	// deploymentMode determines whether the default Alertmanager instance should be deployed
+	// as part of the monitoring stack.
+	// Allowed values are Disabled, DefaultConfig, and CustomConfig.
+	// When set to Disabled, the Alertmanager instance will not be deployed.
+	// When set to DefaultConfig, the platform will deploy Alertmanager with default settings.
+	// When set to CustomConfig, the Alertmanager will be deployed with custom configuration.
+	//
+	// +unionDiscriminator
+	// +required
+	// +kubebuilder:validation:Enum=Disabled;DefaultConfig;CustomConfig
```

Comment:
Nit: Move this marker to the type alias to stay consistent with how you did it for the `LogLevel` type alias.